src/cmd/services/m3comparator/main/querier.go

```diff
@@ -147,15 +155,39 @@ func (q *querier) FetchCompressed(
 	}, cleanup, nil
 }
 
-func (q *querier) generateRandomIters(
+func (q *querier) generateRandomSeries(
 	query *storage.FetchQuery,
-) (encoding.SeriesIterators, error) {
+	blockSize time.Duration,
+) ([]series, error) {
+	var (
+		start = query.Start.Truncate(blockSize)
+		end   = query.End.Truncate(blockSize).Add(blockSize)
+	)
+
+	multiSeriesMetrics := ""
+	for _, matcher := range query.TagMatchers {
+		if bytes.Equal(q.opts.tagOptions.MetricName(), matcher.Name) {
+			if matched, _ := regexp.Match(`^multi_\d+$`, matcher.Value); matched {
+				multiSeriesMetrics = string(matcher.Value)
```

Comment:
nit: this could just do

```
return q.generateMultiSeriesMetrics(string(matcher.Value), start, end, time.Second*30, blockSize)
```

And then can get rid of the `multiSeriesMetrics` variable